### ISSUE TITLE
Avoid installing Python 2 for Mercurial in the commit retrieval image

### ISSUE THREAD

**Issue by marco-c** [mozilla/bugbug#551]
Depends on https://bugzilla.mozilla.org/show_bug.cgi?id=1541166.

**Comment by Lothiraldan** [comment]
Mercurial Python 3 is still beta right now.

**Comment by marco-c** [comment]
Yeah, I wonder if the limited features we are using are working with Python 3. We'll have to check.

**Comment by marco-c** [comment]
@calixteman tested, and this is still not feasible as vct doesn't fully work with Python 3.

**Comment by marco-c** [comment]
I think this would be feasible now, we need to test.

**Comment by marco-c** [comment]
The mozext extension still doesn't support Python 3, so I've had to revert it:
```
hglib.error.ServerError: server exited with status 1: b'** unknown exception encountered, please report by visiting
** https://mercurial-scm.org/wiki/BugTracker
** Python 3.7.4 (default, Oct 17 2019, 06:10:02) [GCC 8.3.0]
** Mercurial Distributed SCM (version 5.2)
** Extensions loaded: purge, strip, robustcheckout, hgmo, pushlog, mozext, firefoxtree, share
Traceback (most recent call last):
  File "/usr/local/bin/hg", line 36, in <module>
    dispatch.run()
  File "/usr/local/lib/python3.7/site-packages/mercurial/dispatch.py", line 111, in run
    status = dispatch(req)
  File "/usr/local/lib/python3.7/site-packages/mercurial/dispatch.py", line 250, in dispatch
    ret = _runcatch(req) or 0
  File "/usr/local/lib/python3.7/site-packages/mercurial/dispatch.py", line 424, in _runcatch
    return _callcatch(ui, _runcatchfunc)
  File "/usr/local/lib/python3.7/site-packages/mercurial/dispatch.py", line 433, in _callcatch
    return scmutil.callcatch(ui, func)
  File "/usr/local/lib/python3.7/site-packages/mercurial/scmutil.py", line 177, in callcatch
    return func()
  File "/usr/local/lib/python3.7/site-packages/mercurial/dispatch.py", line 414, in _runcatchfunc
    return _dispatch(req)
  File "/usr/local/lib/python3.7/site-packages/mercurial/dispatch.py", line 1174, in _dispatch
    lui, repo, cmd, fullargs, ui, options, d, cmdpats, cmdoptions
  File "/usr/local/lib/python3.7/site-packages/mercurial/dispatch.py", line 862, in runcommand
    ret = _runcommand(ui, options, cmd, d)
  File "/usr/local/lib/python3.7/site-packages/mercurial/dispatch.py", line 1185, in _runcommand
    return cmdfunc()
  File "/usr/local/lib/python3.7/site-packages/mercurial/dispatch.py", line 1171, in <lambda>
    d = lambda: util.checksignature(func)(ui, *args, **strcmdopt)
  File "/usr/local/lib/python3.7/site-packages/mercurial/util.py", line 1843, in check
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/mercurial/util.py", line 1843, in check
    return func(*args, **kwargs)
  File "/version-control-tools/hgext/hgmo/__init__.py", line 597, in servehgmo
    return orig(ui, repo, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/mercurial/util.py", line 1843, in check
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/mercurial/commands.py", line 6495, in serve
    return server.runservice(opts, initfn=service.init, runfn=service.run)
  File "/usr/local/lib/python3.7/site-packages/mercurial/server.py", line 164, in runservice
    return runfn()
  File "/usr/local/lib/python3.7/site-packages/mercurial/commandserver.py", line 429, in run
    return sv.serve()
  File "/usr/local/lib/python3.7/site-packages/mercurial/commandserver.py", line 363, in serve
    while self.serveone():
  File "/usr/local/lib/python3.7/site-packages/mercurial/commandserver.py", line 337, in serveone
    handler(self)
  File "/usr/local/lib/python3.7/site-packages/mercurial/commandserver.py", line 321, in runcommand
    ret = dispatch.dispatch(req) & 255
  File "/usr/local/lib/python3.7/site-packages/mercurial/dispatch.py", line 250, in dispatch
    ret = _runcatch(req) or 0
  File "/usr/local/lib/python3.7/site-packages/mercurial/dispatch.py", line 424, in _runcatch
    return _callcatch(ui, _runcatchfunc)
  File "/usr/local/lib/python3.7/site-packages/mercurial/dispatch.py", line 433, in _callcatch
    return scmutil.callcatch(ui, func)
  File "/usr/local/lib/python3.7/site-packages/mercurial/scmutil.py", line 177, in callcatch
    return func()
  File "/usr/local/lib/python3.7/site-packages/mercurial/dispatch.py", line 414, in _runcatchfunc
    return _dispatch(req)
  File "/usr/local/lib/python3.7/site-packages/mercurial/dispatch.py", line 1174, in _dispatch
    lui, repo, cmd, fullargs, ui, options, d, cmdpats, cmdoptions
  File "/usr/local/lib/python3.7/site-packages/mercurial/dispatch.py", line 862, in runcommand
    ret = _runcommand(ui, options, cmd, d)
  File "/usr/local/lib/python3.7/site-packages/mercurial/dispatch.py", line 1185, in _runcommand
    return cmdfunc()
  File "/usr/local/lib/python3.7/site-packages/mercurial/dispatch.py", line 1171, in <lambda>
    d = lambda: util.checksignature(func)(ui, *args, **strcmdopt)
  File "/usr/local/lib/python3.7/site-packages/mercurial/util.py", line 1843, in check
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/mercurial/util.py", line 1843, in check
    return func(*args, **kwargs)
  File "/version-control-tools/hgext/firefoxtree/__init__.py", line 484, in pullcommand
    return orig(ui, repo, source=source, **opts)
  File "/usr/local/lib/python3.7/site-packages/mercurial/util.py", line 1843, in check
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/mercurial/commands.py", line 5453, in pull
    opargs=pullopargs,
  File "/version-control-tools/hgext/mozext/__init__.py", line 587, in wrappedpull
    repo._update_remote_refs(remote, tree)
  File "/version-control-tools/hgext/mozext/__init__.py", line 1460, in _update_remote_refs    if tree not in TREE_ALIASES[\'releases\']:
KeyError: \'releases\''
```

**Comment by marco-c** [comment]
Filed https://bugzilla.mozilla.org/show_bug.cgi?id=1604536.

**Comment by marco-c** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1604536 has been fixed, 1564cd00a749893627028c9343ec63e850c0ba14 is fixing this again. Let's hope it sticks.

**Comment by marco-c** [comment]
Reverted again (with 1361684814165e217b82ac55fc751f5a6032457c) as robustcheckout is failing, and filed https://bugzilla.mozilla.org/show_bug.cgi?id=1604652.
